jax/interpreters/mlir.py

```diff
@@ -1075,7 +1081,17 @@ def _call_lowering(fn_name, stack_name, call_jaxpr, backend, ctx, avals_in,
   output_types = map(aval_to_ir_types, avals_out)
   output_types = [token_type()] * len(effects) + output_types
   flat_output_types = util.flatten(output_types)
-  symbol_name = lower_jaxpr_to_fun(ctx, fn_name, call_jaxpr, effects).name.value
+  if not call_jaxpr.consts:
+    # Cacheable.
+    key = (fn_name, call_jaxpr.jaxpr, tuple(effects))
+    try:
+      func_op = ctx.cached_call_jaxpr_lowerings[key]
+    except KeyError:
+      func_op = lower_jaxpr_to_fun(ctx, fn_name, call_jaxpr, effects)
+      ctx.cached_call_jaxpr_lowerings[key] = func_op
+  else:
+    func_op = lower_jaxpr_to_fun(ctx, fn_name, call_jaxpr, effects)
+  symbol_name = func_op.name.value
   args = [*tokens_in.tokens(), *args]
```

Comment:
Maybe bundle all this maybe-cache-lookup logic up into a separate function, e.g. `lower_jaxpr_to_fun_cached`? Just to keep things tidy.

Reply:
Done.